### PR TITLE
Add more tests for realmadmin

### PR DESCRIPTION
**Pull request by sethvargo**
Also refactor some of the earlier ones to be reusable.

**Release Note**

```release-note
Add more tests for realmadmin
```

### CODE REVIEW

File: pkg/controller/apikey/create.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (c *Controller) HandleCreate() http.Handler {
 		apiKey, err := currentRealm.CreateAuthorizedApp(c.db, &authApp, currentUser)
 		if err != nil {
 			if database.IsValidationError(err) {
-				w.WriteHeader(http.StatusBadRequest)
+				w.WriteHeader(http.StatusUnprocessableEntity)
```
Comment (whaught) [inline pkg/controller/apikey/create.go:69]: Just wondering - when should we be using 422 over 400? We have several places that validate input and mostly go with the 400

Comment (sethvargo) [reply pkg/controller/apikey/create.go:69]: 422 generally means "you gave us data, we got the data, we parsed the data, but the data was invalid for some reason (usually validation)"

400 generall means "we didn't understand the request"

https://www.bennadel.com/blog/2434-http-status-codes-for-invalid-data-400-vs-422.htm

Comment (sethvargo) [reply pkg/controller/apikey/create.go:69]: That being said, we have to keep any _APIs_ at 400 because that's what they're documented as.

**Review by whaught — APPROVED** [review]

**Pull request merged**